### ISSUE TITLE
TextField has cursor position reset to 0 and strange behavior when deleting characters under macOS using cjk input method.

### ISSUE THREAD

**Issue by YeungKC** [flutter/flutter#77052]
code: 

```dart

import 'package:flutter/material.dart';

void main() => runApp(
      MaterialApp(
        home: Scaffold(
          body: Center(
            child: TextField(),
          ),
        ),
      ),
    );

```

result: 

https://user-images.githubusercontent.com/11473691/109631945-73479200-7b81-11eb-9910-a4b24fefcc50.mp4

flutter --version
Flutter 1.27.0-9.0.pre.149 • channel master • https://github.com/flutter/flutter.git
Framework • revision 9964e8fe38 (6 小时前) • 2021-03-01 23:00:02 -0500
Engine • revision 17f6bfc480
Tools • Dart 2.13.0 (build 2.13.0-90.0.dev)

**Comment by YeungKC** [comment]
btw, it seems that the input method is not located near the TextField.

**Comment by GreenLim** [comment]
And another related problem is the input (tested in Chinese input) will broken  Ctrl-A behavior.
It seems those problem are related to this pr https://github.com/flutter/engine/pull/24533.

**Comment by cbracken** [comment]
@GreenLim can you file a new issue and provide repro steps for the ctrl-A behaviour (then CC me on it so I can take a look)?  If you have bisected it to flutter/engine#24533 then please include that information in that bug. There have been a lot of other changes recently related to text input, particularly around non-printing keypresses such as ctrl/shift/etc. modifiers.

**Comment by cbracken** [comment]
I'm going to close this as a duplicate of https://github.com/flutter/flutter/issues/30660 since CJK and IME support is still a work in progress on macOS and not landed.

**Comment by GreenLim** [comment]
> @GreenLim can you file a new issue and provide repro steps for the ctrl-A behaviour (then CC me on it so I can take a look)? If you have bisected it to [flutter/engine#24533](https://github.com/flutter/engine/pull/24533) then please include that information in that bug. There have been a lot of other changes recently related to text input, particularly around non-printing keypresses such as ctrl/shift/etc. modifiers.

I can test Chinese input after #30660 was closed. If it remains the problem of ctrl-A and other strange behaviors, I will file a new issue then CC you on it. Thanks for your work.
